google/cloud/aiplatform/jobs.py

```diff
@@ -344,7 +344,7 @@ def completion_stats(self) -> Optional[gca_completion_stats.CompletionStats]:
     @classmethod
     def create(
         cls,
-        job_display_name: str,
+        job_display_name: Optional[str],
```

Comment:
Preference to remove this pattern as valid usage would require passing `None` explicitly.

Please remove throughout where `display_name` is a required argument and cannot be changed to have a default argument because it breaks order of arguments.

Reply:
This is a type annotation. Type annotations specify what argument types are valid for a parameter.
In Python `Optional[T]` is the same as `Union[T, None]`. In Google codebases I've also seen tha `Optional[str]` means `str or None`. Such annotation states that `None` is a valid argument for a parameter. Otherwise, when the user passes `display_name=None`, some type checkers can complain that `None` is not a valid argument type for this parameter.

Reply:
Understand. My preference is to not recommend users pass `None` explicitly as a workaround to our inability to change the argument ordering. This could have downstream effect on samples, notebooks, and other ancillary artifacts being created with this type of usage:

```
job = BatchPredictionJob.create(job_display_name=None, model_name='projects/.../my_model')
``` 

Which, to me, seems to indicate that display_name is not set.

Perhaps we can update the docstrings to indicate the format of the name when passed `None`.

@morgandu @kweinmeister Thoughts? If you think this usage is OK then we can merge in. I'm open to changing this view.

Reply:
I have not seen this style of explicitly passing in `None`, and then having the value overridden in the function.

In the example above, anyone reading the code without knowledge of the `BatchPredictionJob` internals would assume that the display name is set to `None`.

I would suggest we start with applying this change only to functions where we can specify `Optional[str] = None`. It will have a clear usability benefit. We can always come back later with another PR for cases in which `display_name` is required, after we have given it more thought.

Reply:
Having `Optional[T]` parameter without a default value is pretty common even in the Vertex SDK.

However I can remove the annotation changes as asked.

https://github.com/googleapis/python-aiplatform/blob/89e321246b6223a2355947d8dbd0161b84523478/google/cloud/aiplatform/initializer.py#L102

https://github.com/googleapis/python-aiplatform/blob/5735545c647075d00c0d18f4a3319c76be8dc110/google/cloud/aiplatform/models.py#L327

https://github.com/googleapis/python-aiplatform/blob/5735545c647075d00c0d18f4a3319c76be8dc110/google/cloud/aiplatform/models.py#L328

https://github.com/googleapis/python-aiplatform/blob/ac0c4e79bc2882480f5e7a0dc4d8c051a6759bbf/google/cloud/aiplatform/training_jobs.py#L1676

https://github.com/googleapis/python-aiplatform/blob/ac0c4e79bc2882480f5e7a0dc4d8c051a6759bbf/google/cloud/aiplatform/training_jobs.py#L2134

https://github.com/googleapis/python-aiplatform/blob/ac0c4e79bc2882480f5e7a0dc4d8c051a6759bbf/google/cloud/aiplatform/training_jobs.py#L3267

https://github.com/googleapis/python-aiplatform/blob/81c403a6b833aae3836f3b3a98f8f09602839c05/google/cloud/aiplatform/models.py#L352

https://github.com/googleapis/python-aiplatform/blob/93acea098e01cc9931a403037c309217cd391be2/google/cloud/aiplatform/training_jobs.py#L557

https://github.com/googleapis/python-aiplatform/blob/57b12412ff5b79bf5819f82cd807491a48cfcf93/google/cloud/aiplatform/datasets/dataset.py#L257

https://github.com/googleapis/python-aiplatform/blob/b613b264524aaab2cb65e63a5487770736faa7c8/google/cloud/aiplatform/tensorboard/plugins/tf_profiler/profile_uploader.py#L81

Reply:
As asked, I've removed the `Optional[str]` annotation and docstring changes in cases where specifying the default value is not possible.